include/pcmki/pcmki_transition.h

```diff
@@ -28,19 +28,39 @@ typedef enum {
 typedef struct te_timer_s crm_action_timer_t;
 typedef struct crm_graph_s crm_graph_t;
 
+enum pcmk__synapse_flags {
+    pcmk__synapse_ready       = (1 << 0),
+    pcmk__synapse_failed      = (1 << 1),
+    pcmk__synapse_executed    = (1 << 2),
+    pcmk__synapse_confirmed   = (1 << 3),
+};
+
 typedef struct synapse_s {
     int id;
     int priority;
 
-    gboolean ready;
-    gboolean failed;
-    gboolean executed;
-    gboolean confirmed;
+    uint32_t flags; // Group of pcmk__synapse_flags
 
     GList *actions;           /* crm_action_t* */
     GList *inputs;            /* crm_action_t* */
 } synapse_t;
 
+const char *synapse_state_str(synapse_t *synapse);
+
+#define pcmk__set_synapse_flags(synapse, flags_to_set) do {             \
+        (synapse)->flags = pcmk__set_flags_as(__func__, __LINE__,       \
+            LOG_TRACE,                                                  \
+            "Synapse", synapse_state_str(synapse),                       \
+            (synapse)->flags, (flags_to_set), #flags_to_set);           \
```

Comment:
FYI, to explain another obscure usage: in a #define macro, #varname is the literal string of what was passed as varname. For example, assume a call like

    pcmk__set_synapse_flags(synapse, pcmk__synapse_confirmed);

If we only passed flags_to_set, then pcmk__set_flags_as() could only log the numeric value of the flags, like

    Synapse flags 0x00000100 set by pcmk_sched_transition.c:599

which is a pain to read because you have to look up what the possible flags are. #flags_to_set is the literal string "pcmk__synapse_confirmed" in this case, so with that we can log

    Synapse flags 0x00000100 (pcmk__synapse_confirmed) set by pcmk_sched_transition.c:599

which is much easier to understand at a glance.